tibber_aws/aws_logs.py

```diff
@@ -0,0 +1,149 @@
+import logging
+from dataclasses import dataclass
+from datetime import datetime as dt
+import json
+from .aws_base import AwsBase
+from typing import Dict, List
+from types_aiobotocore_logs.client import CloudWatchLogsClient
+
+logger = logging.getLogger("aws_logs")
+
+
+@dataclass
+class CloudWatchLogEvent:
+    """An AWS CloudWatch log event (a log line)"""
+    logStreamName: str
+    timestamp: int
+    message: str
+    ingestionTime: int
+    eventId: str
+
+    def __post_init__(self):
+        self.timestamp_dt = dt.fromtimestamp(self.timestamp / 1000)
+        self.ingestionTime_dt = dt.fromtimestamp(self.ingestionTime / 1000)
+        self.message = json.loads(self.message)
+
+
+class Logs(AwsBase):
+    """CloudWatch Logs client to query AWS logs from a log stream."""
+    _client: CloudWatchLogsClient
+
+    def __init__(self, region_name="eu-west-1") -> None:
+        super().__init__("logs", region_name)
+
+    async def _get_log_events_raw(
+        self, event: str, log_group: str, start_time: dt, end_time: dt, extra_filter: dict = None, **kwargs
```

Comment:
`event` is specific to structlog, not something general that's always there, better have callers specify this in the filter if they care about filtering on it

Reply:
That's why it's called `get_log_events`, but noticed that the `CloudWatchLogsClient` actually have a method called `get_log_events` which is an un-filtered version. So point made! I can expose a more of the raw `CloudWatchLogsClient`s to enable more freedom.

I'll rename this to `get_struclog_events`.